public/app/plugins/panel/graph/graph.ts

```diff
@@ -428,7 +452,9 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
         let defaultTicks = panelWidth / 50;
 
         if (data.length && bucketSize) {
-          ticks = _.map(data[0].data, point => point[0]);
+          ticks = _.chain(data).map(d => {
+            return _.map(d.data, point => point[0]);
```

Comment:
Sorry, I didn't know the map/each is slow. I compare map and for. Actually, map is slower than for.
https://jsperf.com/map-uniq-vs-for/1

At this line, collect all histogram x-axis value from multiple histogram data.
point[0] is histogram x-axis value, so the total count is `bucketSize x the number of time series`.